transport/grpc/options.go

```diff
@@ -166,6 +165,13 @@ func ContextDialer(f func(context.Context, string) (net.Conn, error)) DialOption
 	}
 }
 
+// Compressor sets the compressor to be used by default for gRPC connections
+func Compressor(compressor string) DialOption {
```

Comment:
I dislike that we have to use a string here, but I don/t know whether we have better options.

This is surfacing an API wart of gRPC-go that we might have to emulate if we pivoted the implementation to x/net/http2.

If we were to do this from scratch, we would certainly have this call option accept a factory for the type of compressor this outbound should use, not a reference to one through a string.

We could introduce a `yarpc.Compression` type that has a `compression.Name()string` method and a `compression.New()yarpc.Compressor` method that returns an instance of the compressor for a request.

That would make this interface forward compatible if we chose to pivot the underlying grpc implementation to x/net/http2 in the future. For now, you would just call Name() and throw away the implementation.